bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -210,8 +211,20 @@ protected void resetExecutionContext(ScriptEngine engine, Map<String, ?> context
     protected @Nullable Object eval(ScriptEngine engine, String script) {
         try {
             if (compiledScript.isPresent()) {
+                if (engine instanceof Lock lock) {
+                    lock.lock();
```

Comment:
Wouldn't it make sense to use `tryLock(1, TimeUnit.MINUTES)` here and abort with an ERROR log? That would avoid blocking too many threads when the lock could not be obtained.

Reply:
Absolutely!
I will change that 👍